cpp/locality/AABBQuery.h

```diff
@@ -101,26 +101,31 @@ class AABBQuery : public NeighborQuery
     AABBTree m_aabb_tree; //!< AABB tree of points
 
 protected:
+    //! Validate the combination of specified arguments.
+    /*! Override parent function to account for the various arguments
+     *  specifically required for AABBQuery nearest neighbor queries.
+     */
     virtual void validateQueryArgs(QueryArgs& args) const
     {
+        inferMode(args);
         if (args.mode == QueryArgs::ball)
         {
-            if (args.rmax == -1)
-                throw std::runtime_error("You must set rmax in the query arguments.");
+            if (args.r_max == QueryArgs::DEFAULT_R_MAX)
+                throw std::runtime_error("You must set r_max in the query arguments when performing ball queries.");
         }
         else if (args.mode == QueryArgs::nearest)
         {
-            if (args.nn == -1)
-                throw std::runtime_error("You must set nn in the query arguments.");
-            if (args.scale == -1)
+            if (args.num_neighbors == QueryArgs::DEFAULT_NUM_NEIGHBORS)
+                throw std::runtime_error("You must set num_neighbors in the query arguments when performing number of neighbor queries.");
+            if (args.scale == QueryArgs::DEFAULT_SCALE)
             {
                 args.scale = float(1.1);
             }
-            if (args.rmax == -1)
+            if (args.r_max == QueryArgs::DEFAULT_R_MAX)
             {
                 vec3<float> L = this->getBox().getL();
-                float rmax = std::min(L.x, L.y);
-                args.rmax = this->getBox().is2D() ? float(0.1) * rmax : float(0.1) * std::min(rmax, L.z);
+                float r_max = std::min(L.x, L.y);
+                args.r_max = this->getBox().is2D() ? float(0.1) * r_max : float(0.1) * std::min(r_max, L.z);
```

Comment:
Make the magic constant `float(0.1)` only appear one time.
```suggestion
                if (this->getBox().is2D())
                    r_max = std::min(r_max, L.z);
                args.r_max = float(0.1) * r_max;
```